src/rmm.cpp

```diff
@@ -54,32 +57,36 @@ const char * rmmGetErrorString(rmmError_t errcode) {
 // Initialize memory manager state and storage.
 rmmError_t rmmInitialize(rmmOptions_t *options)
 {
+
     rmm::Manager::getInstance().initialize(options);
+    rmm::mr::device_memory_resource * memory_resource;
 
     if (rmm::Manager::usePoolAllocator())
     {
-        cnmemDevice_t dev;
-        RMM_CHECK_CUDA( cudaGetDevice(&(dev.device)) );
-        // Note: cnmem defaults to half GPU memory
-        dev.size = rmm::Manager::getOptions().initial_pool_size; 
-        dev.numStreams = 1;
-        cudaStream_t streams[1]; streams[0] = 0;
-        dev.streams = streams;
-        dev.streamSizes = 0;
-        unsigned flags = rmm::Manager::useManagedMemory() ? CNMEM_FLAGS_MANAGED : 0;
-        RMM_CHECK_CNMEM( cnmemInit(1, &dev, flags) );
+        memory_resource = static_cast<rmm::mr::device_memory_resource *>(
+            new rmm::mr::cnmem_memory_resource(rmm::Manager::getOptions().initial_pool_size));
+
+    }else if(rmm::Manager::useManagedMemory()){
+      memory_resource = static_cast<rmm::mr::device_memory_resource *>(
+          new rmm::mr::managed_memory_resource());
+
+    }else{
+        memory_resource =  static_cast<rmm::mr::device_memory_resource *>(
+            new rmm::mr::cuda_memory_resource());
     }
+    rmm::mr::device_memory_resource * old_resource = rmm::mr::set_default_resource(memory_resource);
+    //TODO: Talk to Mark and Jake about how this is a bit weird and will have leaks
```

Comment:
Instead of dynamically allocating the new resources, just add a static instance of each resource via the same "[Construct on First Use](https://isocpp.org/wiki/faq/ctors#static-init-order)" mechanism I did for the [`initial_resource`](https://github.com/rapidsai/rmm/blob/branch-0.10/src/mr/default_memory_resource.cpp#L26)

So you'd add a new function for each resource, e.g., 
```
device_memory_resource* pool_resource(std::size_t pool_size = 0){
   static cnmem_memory_resource mr{pool_size};
   return &mr;
}
```

There should be one of these functions for every resource that RMM provides by default, e.g., pool, cuda, and managed.

Reply:
Can do. are you saying the pool_size should no longer default to half of the gpu memory as it used to?

Reply:
nvm i see thats set elsewhere